### DISCUSSION TITLE
Ortur Laser Master 2 + ESP8266 for WiFi

### DISCUSSION THREAD

**Discussion by Apology11** [arkypita/LaserGRBL#1350] [Q&A] [answered]
Hello everyone,

does somebody here have experience connecting the Ortur Laser Master 2 to an ESP8266 board to use LaserGRBL with WiFi, as described here:
https://lasergrbl.com/usage/wifi-with-esp8266/

On the Ortur Laser Master 2 mainboard, there is a connector for an external panel, which has ports labeled RXD and TXD.
However, I tried connecting to them, but they do not send serial data. 
It also seems that the USB port is connected to the STM32 chip directly, so that the serial connection cannot be intercepted.
Has anyone tried this before or has any suggestions?

**Comment by Apology11** [comment]
Here an image of the backside of the mainboard and the cables attached.

![ortuerlm2](https://user-images.githubusercontent.com/3321529/108636223-2f1f0800-7484-11eb-8e5f-e5524c46c0cb.JPG)

**Comment by arkypita** [comment, marked as answer]
I ask something to Ortur support some month ago
This is the reply.

> Actual firmware isn’t “listening” to the connections because of firmware size issue. However a new firmware with optimized memory size that will have a serial interface is under development.

**Reply by arkypita** [reply]
PS. I try to do the same, without success.

STM32 has internal USB support, so the only way is to have a "usb host" chip that fake the USB connection from external USB plug, or wait for serial-enabled firmware, or use an open source STM32 grbl firmware.

Unlucky Ortur does not release their firmware as open source license require.

**Reply by Apology11** [reply]
Thank you for the feedback!
I could not find any information about this before.
Did you try this with Firmware 1.4? Because mine is still at 1.37.

If it does not work on 1.40, then I will consider using a Raspberry Pi Zero W, connect the laser via USB-OTG adapter and port your WebsocketServer Code to Python, to run it on there....

**Reply by Apology11** [reply]
I implemented a Serial <-> Websocket Bridge in node.js, which can run on Linux/Windows/Mac inclduing a Raspberry Pi Zero W or similar Boards and it works.
It lacks the commands to reconfigure Wifi, but I think this is not necessary on boards with a real OS.
If somebody is interested:
https://github.com/Apology11/SerialWebsocketJS.git

**Reply by arkypita** [reply]
thank you

**Reply by j-lang** [reply]
In simple words:
A raspberry PI Zero W can act as the Websocket bridge similar to the ESP8266 solution, where the ortur laser is connected via USB?
